### PR TITLE
Fix the missing ninja in AARCH64 Linux wheel recipe

### PR DESCRIPTION
**Pull request by maliasadi**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
- Fix the missing `ninja` in Linux AARCH64 wheels recipe (Fix the issue reported [here](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/12031173297/job/33547598917))
- Upgrade the Kokkos version support to [4.5.00](https://github.com/kokkos/kokkos/releases/tag/4.5.00).  

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-79255]

### CODE REVIEW

**Review by multiphaseCFD — APPROVED** [review]
LGTM! Thanks @maliasadi !

File: .github/workflows/wheel_linux_aarch64.yml [hunk 1]
```diff
@@ -75,8 +75,9 @@ jobs:
             -i ${{ matrix.container_img }} \
             bash -c "git config --global --add safe.directory /io && \
             cd /io && \
-            python3.10 -m pip install ninja && \
-            ln -s /opt/python/cp310-cp10/bin/ninja /usr/bin/ninja && \
+            echo $(which python3.10) && \
```
Comment (LuisAlfredoNu) [inline .github/workflows/wheel_linux_aarch64.yml:78]: Do we still need to print this and verbose ninja or was it for debugging purposes?  :thinking:

**Review by LuisAlfredoNu — COMMENTED** [review]
LGTM. Thanks @maliasadi 
Just one comment :eyes:

File: .github/workflows/wheel_linux_aarch64.yml [hunk 2]
```diff
@@ -159,6 +155,8 @@ jobs:
 
           CIBW_BEFORE_TEST: |
             python -m pip install -r requirements-tests.txt
+            # FIXME: install PennyLane/latest to fix pennylane.devices.capabilities requirements
```
Comment (josephleekl) [inline .github/workflows/wheel_linux_aarch64.yml:158]: Will we remove these two lines after the next release?

**Comment by AmintorDusko** [comment]
The error is still there. Python can't find the binaries.

File: .github/workflows/wheel_linux_aarch64.yml [hunk 3]
```diff
@@ -123,16 +124,11 @@ jobs:
           mkdir Kokkos
           cp -rf ${{ github.workspace }}/Kokkos_install/${{ matrix.exec_model }}/* Kokkos/
 
-      - name: Install Python 3.10
-        uses: actions/setup-python@v5
-        with:
-          python-version: '3.10'
-
```
Comment (AmintorDusko) [inline .github/workflows/wheel_linux_aarch64.yml:130]: This action usually helps with PATHs etc. You can try to bring it back and see if this solves the problem.
GH native python can be a little unreliable sometimes.

File: .github/workflows/wheel_linux_aarch64.yml (hunk 2 above)
Comment (maliasadi) [reply .github/workflows/wheel_linux_aarch64.yml:163]: Yes!
```suggestion
            # FIXME(v0.40.0): install PennyLane/latest to fix pennylane.devices.capabilities requirements
```

**Comment by maliasadi** [comment]
Building both Kokkos and Lightning-Kokkos with GCC-13 fixed [the last issue](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/12075397180/job/33675157946?pr=1007) on Linux AARCH64.

**Review by LuisAlfredoNu — APPROVED** [review]
Thank you @maliasadi 
Good work!

**Pull request merged**